### PR TITLE
Added the broken links cron

### PR DESCRIPTION
**Pull request by danielborza**

### CODE REVIEW

File: src/Helpers/ManualHelper.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: externaldev
+ * Date: 21.01.19
+ * Time: 09:37
+ */
+
+namespace PlentyManual\Helpers;
```
Comment (felixdausch) [inline src/Helpers/ManualHelper.php]: why is this class required if it only passes data to an existing function?

File: src/Services/BrokenLinks.php [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+    private $manualHelper;
+
+    const RECIPIENTS = [
+        'doku@plentymarkets.com'
+    ];
+
+    const RECIPIENT_DEV = [
+        'daniel.borza@plentydevelopment.com'
+    ];
```
Comment (felixdausch) [inline src/Services/BrokenLinks.php]: it might be usefull to put this to the plugin config

File: src/Services/BrokenLinks.php [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+        return $result;
+    }
+
+    /**
+     * @param $resultData
+     * @param $lang
+     * @return array
+     */
+    private function Links($resultData, $lang)
```
Comment (felixdausch) [inline src/Services/BrokenLinks.php]: method names should be lower case

File: src/Services/BrokenLinks.php [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+                ]
+            ],
+            "size" => 1000
+        ];
+
+        $type = "manual_".$lang;
+
+        $requestArray = array('host' => $this->host,
+            'type' => $type,
+            'Query' => $requestQuery);
```
Comment (felixdausch) [inline src/Services/BrokenLinks.php]: why do you define keys 'host' and 'type' (lowercase) and a key 'Query' (uppercase)? Please use lowercase keys

File: src/Services/BrokenLinks.php [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+        return $result;
+    }
+
+    /**
+     * @param $lang
+     * @return array
+     */
+    private function brokenLinksArray($lang)
+    {
+        $requestQuery = [
```
Comment (felixdausch) [inline src/Services/BrokenLinks.php]: could you please explain what you are doing here? Broken links are not stored in the elasticsearch index?

Comment (danielborza) [reply src/Services/BrokenLinks.php]: In this function, I search for all the URLs from elasticsearch than I create one array with all the URLs. After that, I test each URL if there is a path that matches that URL in the Matadata.php, if there is no path we display the error 404, so this function should show links that are changed or it should be deleted from elasticsearch

File: src/Helpers/ManualHelper.php (hunk 1 above)
Comment (danielborza) [reply src/Helpers/ManualHelper.php]: I removed the class, you were right it was not necessary to have it

File: src/Services/BrokenLinks.php (hunk 2 above)
Comment (danielborza) [reply src/Services/BrokenLinks.php]: Added the email to the plugin config

File: src/Services/BrokenLinks.php (hunk 3 above)
Comment (danielborza) [reply src/Services/BrokenLinks.php]: Changed the method name

File: src/Services/BrokenLinks.php (hunk 4 above)
Comment (danielborza) [reply src/Services/BrokenLinks.php]: Changed the 'Query' to 'query'

**Pull request closed without merging**